### PR TITLE
Invalidate saved check-ins

### PR DESCRIPTION
**Pull request by ffactory-ofcl**
* invalidate saved check-in after 1 hour, even if the page stays open
* delete old & expired checkins
* allow to be checked in at multiple locations (would make sense if the building is a location and the rooms within it are locations)
* code cleanup

### CODE REVIEW

File: src/serverMain/kotlin/com/studo/campusqr/endpoints/UserFrontend.kt [hunk 1]
```diff
@@ -51,15 +51,25 @@ suspend fun ApplicationCall.userFrontend() {
...
           }
-          +LocalizedString(
-            "Please close this page and scan the QR code again.",
-            "Bitte schließen Sie dieses Fenster und scannen Sie den QR Code erneut."
-          ).get(this@userFrontend)
+          span {
+            id = "overlay-retry"
+            +LocalizedString(
+              "Please close this page and scan the QR code again.",
+              "Bitte schließen Sie dieses Fenster und scannen Sie den QR Code erneut."
```
Comment (vonox7) [inline src/serverMain/kotlin/com/studo/campusqr/endpoints/UserFrontend.kt:63]:
```suggestion
              "Bitte schließen Sie diese Seite und scannen Sie den QR Code erneut."
```

File: src/serverMain/kotlin/com/studo/campusqr/endpoints/UserFrontend.kt [hunk 2]
```diff
@@ -51,15 +51,25 @@ suspend fun ApplicationCall.userFrontend() {
...
+            ).get(this@userFrontend)
+          }
+          span("hidden") {
+            id = "overlay-expired"
+            +LocalizedString(
+              "Your check-in has expired. Please close this page and scan the QR code again.",
+              "Ihr Check-in ist abgelaufen. Bitte schließen Sie dieses Fenster und scannen Sie den QR Code erneut."
```
Comment (vonox7) [inline src/serverMain/kotlin/com/studo/campusqr/endpoints/UserFrontend.kt:70]:
```suggestion
              "Ihr Check-in ist abgelaufen. Bitte schließen Sie diese Seite und scannen Sie den QR Code erneut."
```

File: src/serverMain/resources/userFrontend/userFrontend.js [hunk 3]
```diff
@@ -87,7 +85,7 @@ function normalStartup() {
               // Success
               setDatetimeElement(Date.now())
               window.localStorage.setItem("email", email);
-              window.localStorage.setItem("checkin", `${btoa(email)}::${locationId}::${Date.now().toString()}`);
+              window.localStorage.setItem("checkin-" + locationId, `${btoa(email)}::${Date.now().toString()}`);
```
Comment (vonox7) [inline src/serverMain/resources/userFrontend/userFrontend.js:88]: Good catch!

**Review by vonox7 — APPROVED** [review]

File: src/serverMain/resources/userFrontend/userFrontend.js (hunk 3 above)
Comment (ffactory-ofcl) [reply src/serverMain/resources/userFrontend/userFrontend.js:88]: Thanks ... But what do you mean by that 😅

Comment (vonox7) [reply src/serverMain/resources/userFrontend/userFrontend.js:88]: That you could have 2 concurrent checkins in 2 locations

**Pull request merged**